pkg/apis/kops/validation/validation.go

```diff
@@ -600,6 +595,25 @@ func validateContainerRuntime(runtime *string, fldPath *field.Path) field.ErrorL
 	return allErrs
 }
 
+func validateDockerConfig(config *kops.DockerConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if config.Version != nil {
+		valid := []string{"17.03.2", "17.09.0", "18.03.1", "18.06.1", "18.06.2", "18.06.3", "18.09.3", "18.09.9", "19.03.4", "19.03.8"}
+		allErrs = append(allErrs, IsValidValue(fldPath.Child("version"), config.Version, valid)...)
```

Comment:
This validation is the critical thing to backport, I believe.  (I think the others might have a higher risk of collisions).

Should we split this out into its own commit that we cherry-pick back?  And I'm wondering whether it's easier just to exclude the known-bad versions (1.11, 1.12 etc) with an explicit "docker removed these versions" message...  Technically I guess it doesn't really matter if we let through 19.03.8 on an older version that doesn't support it - we're no worse off than we are today.